compiler/src/iree/compiler/Codegen/WGSL/test/replace_push_constants.mlir

```diff
@@ -0,0 +1,89 @@
+func.func @constantLoadIndex() {
+  // CHECK: %[[SUBSPAN:.+]] = hal.interface.binding.subspan set(3) binding(0) type(storage_buffer) offset(%c0) : !flow.dispatch.tensor<readonly:i32>
+  // CHECK: %[[LOAD:.+]] = flow.dispatch.tensor.load %[[SUBSPAN]], offsets = [], sizes = [], strides = [] : !flow.dispatch.tensor<readonly:i32> -> tensor<1xi32>
+  // CHECK: %[[EXTRACT:.+]] = tensor.extract %[[LOAD]][%c0_0] : tensor<1xi32>
+  // CHECK: %[[CAST:.+]] = arith.index_cast %[[EXTRACT]] : i32 to index
+  %0 = hal.interface.constant.load[0] : index
+  // CHECK: = arith.index_cast %[[CAST]] : index to i32
+  %1 = arith.index_cast %0 : index to i32
+  return
+}
+
+// -----
+
+// CHECK-LABEL: @constantLoadI32
+func.func @constantLoadI32() {
+  // CHECK: %[[SUBSPAN:.+]] = hal.interface.binding.subspan set(3) binding(0) type(storage_buffer) offset(%c0) : !flow.dispatch.tensor<readonly:i32>
+  // CHECK: %[[LOAD:.+]] = flow.dispatch.tensor.load %[[SUBSPAN]], offsets = [], sizes = [], strides = [] : !flow.dispatch.tensor<readonly:i32> -> tensor<1xi32>
+  // CHECK: %[[EXTRACT:.+]] = tensor.extract %[[LOAD]][%c0_0] : tensor<1xi32>
+  // CHECK: %[[CAST:.+]] = arith.bitcast %[[EXTRACT]] : i32 to i32
+  %0 = hal.interface.constant.load[0] : i32
+  // CHECK: = math.absi %[[CAST]] : i32
+  %1 = math.absi %0 : i32
+  return
+}
+
+// -----
+
+// CHECK-LABEL: @constantLoadF32
+func.func @constantLoadF32() {
+  // CHECK: %[[SUBSPAN:.+]] = hal.interface.binding.subspan set(3) binding(0) type(storage_buffer) offset(%c0) : !flow.dispatch.tensor<readonly:i32>
+  // CHECK: %[[LOAD:.+]] = flow.dispatch.tensor.load %[[SUBSPAN]], offsets = [], sizes = [], strides = [] : !flow.dispatch.tensor<readonly:i32> -> tensor<1xi32>
+  // CHECK: %[[EXTRACT:.+]] = tensor.extract %[[LOAD]][%c0_0] : tensor<1xi32>
+  // CHECK: %[[CAST:.+]] = arith.bitcast %[[EXTRACT]] : i32 to f32
+  %0 = hal.interface.constant.load[0] : f32
+  // CHECK: = math.absf %[[CAST]] : f32
+  %1 = math.absf %0 : f32
+  return
+}
+
+// -----
+
+// CHECK-LABEL: @constantLoadWithIndexAndAlignment
+func.func @constantLoadWithIndexAndAlignment() {
+  // CHECK: %[[SUBSPAN:.+]] = hal.interface.binding.subspan set(3) binding(0) type(storage_buffer) offset(%c5) alignment(16) : !flow.dispatch.tensor<readonly:i32>
+  // CHECK: %[[LOAD:.+]] = flow.dispatch.tensor.load %[[SUBSPAN]], offsets = [], sizes = [], strides = [] : !flow.dispatch.tensor<readonly:i32> -> tensor<6xi32>
+  // CHECK: %[[EXTRACT:.+]] = tensor.extract %[[LOAD]][%c5_0] : tensor<6xi32>
+  // CHECK: %[[CAST:.+]] = arith.index_cast %[[EXTRACT]] : i32 to index
+  %0 = hal.interface.constant.load[5] alignment(16) : index
+  // CHECK: = arith.index_cast %[[CAST]] : index to i32
+  %1 = arith.index_cast %0 : index to i32
+  return
+}
+
+// -----
+
+// CHECK-LABEL: @constantLoadMultiple
+func.func @constantLoadMultiple() {
+  // CHECK: %[[SUBSPAN:.+]] = hal.interface.binding.subspan set(3) binding(0) type(storage_buffer) offset(%c2) : !flow.dispatch.tensor<readonly:i32>
```

Comment:
the dispatch tensor also needs to be 3xi32 here (I should probably add a verifier for that)